tiledb/sm/misc/type_casts.h

```diff
@@ -0,0 +1,146 @@
+/**
+ * @file type_casts.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file contains safe type casting utilities.
+ */
+
+#ifndef TILEDB_TYPE_CASTS_H
+#define TILEDB_TYPE_CASTS_H
+
+#include <limits>
+#include <stdexcept>
+
+#include "tiledb/common/unreachable.h"
+#include "tiledb/sm/enums/datatype.h"
+#include "tiledb/sm/misc/types.h"
+
+namespace tiledb::sm::utils {
+
+/**
+ * Safely convert integral values between different bit widths checking for
+ * invalid conversions. The basic checks are just to make sure that the
+ * conversion is roundtrip-able without an intermediate change in sign.
+ *
+ * This would likely be significantly faster if I were to take the time and
```

Comment:
NIT: rephrase to get rid of I here a and lower in the comment.

Reply:
Fixed.